config/config.go

```diff
@@ -217,7 +216,7 @@ type ScrapeConfig struct {
 	// Indicator whether the scraped metrics should remain unmodified.
 	HonorLabels bool `yaml:"honor_labels,omitempty"`
 	// A set of query parameters with which the target is scraped.
-	Params url.Values `yaml:"params,omitempty"`
+	Params map[string]string `yaml:"params,omitempty"`
```

Comment:
That doesn't address the points I raised in my last comment.

So then __param is the last occurrences, __param2 the first one? This is
just vague and continuing adding features this way will leave us with
fragmentation all over the place in the long run.

Right now, the way we are adding new functionality is not aligned with our
stability concerns when extending/modifying them.

On Wed, Jul 29, 2015, 5:09 PM Brian Brazil <notifications@github.com> wrote:

> In config/config.go
> <https://github.com/prometheus/prometheus/pull/934#discussion_r35770676>:
>
> > @@ -217,7 +216,7 @@ type ScrapeConfig struct {
> >  	// Indicator whether the scraped metrics should remain unmodified.
> >  	HonorLabels bool `yaml:"honor_labels,omitempty"`
> >  	// A set of query parameters with which the target is scraped.
> > -	Params url.Values `yaml:"params,omitempty"`
> > +	Params map[string]string `yaml:"params,omitempty"`
>
>  Cannot really check right now, but are there no characters that cannot
> be percent-encoded that could serve as a separator?
>
> The value could be anything.
>
> Let's think this through or explicitly be open to do something breaking in
> the future
>
> This change is non-breaking now, and we can do __param2, __param3 etc. if
> anyone wants to touch more than one param in the future (which I can't
> think of a use case for at the moment).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/prometheus/prometheus/pull/934/files#r35770676>.
>